src/components/Executions/filters/useCurrentUserOnlyFilterState.ts

```diff
@@ -0,0 +1,43 @@
+import { FilterOperationName } from 'models/AdminEntity/types';
+import { UserProfile } from 'models/Common/types';
+import { useState } from 'react';
+import { BooleanFilterState } from './types';
+import { useFilterButtonState } from './useFilterButtonState';
+import { useUserProfile } from 'components/hooks/useUserProfile';
+
+/** Maintains the state for a button to be used for filtering by user.
```

Comment:
Small nit: I think this doc string goes above line 14 instead? That way, code editors which render documentation will show this string whenever a user hovers over a usage of `useCurrentUserOnlyFilterState`